### PR TITLE
[s] xenobio golems can no longer be used to exploit eldritch affairs mechanics

### PR DESCRIPTION
**Pull request by DeltaFire15**
## About The Pull Request
As of now, people could create golems as an traitor cultist, which became actual full-on cultists, allowing for skipping any and all inbuilt restrictions of eldritch affairs. This fixes that so the golems are neutered too. (Done via neutered/traitor so they don't show up on the endround screen either.)

## Why It's Good For The Game
Maaybe having golems / simillar be able to fully circumvent all restrictions set by eldritch affairs aiiin't the best idea..

## Changelog
:cl:
fix: Golems / simillar now inherit the neutered antag datum if the creator is neutered.
/:cl:

### CODE REVIEW

**Comment by Vizoplez** [comment]
@DeltaFire15 this doesnt stop the exact same exploit with narsian golems though :eyes:

File: code/datums/mind.dm [hunk 1]
```diff
@@ -330,15 +330,19 @@
 //Link a new mobs mind to the creator of said mob. They will join any team they are currently on, and will only switch teams when their creator does.
 
 /datum/mind/proc/enslave_mind_to_creator(mob/living/creator)
-	if(iscultist(creator))
+	if(iscultist(creator, TRUE))
 		SSticker.mode.add_cultist(src)
+	else if(iscultist(creator))
+		src.add_antag_datum(/datum/antagonist/cult/neutered/traitor)
```
Comment (Ghommie) [inline code/datums/mind.dm]:
```suggestion
	if(iscultist(creator))
		if(iscultist(creator, TRUE))
			SSticker.mode.add_cultist(src)
		else
			add_antag_datum(/datum/antagonist/cult/neutered/traitor)
```

File: code/datums/mind.dm [hunk 2]
```diff
@@ -330,15 +330,19 @@
...
 
 	else if(is_revolutionary(creator))
 		var/datum/antagonist/rev/converter = creator.mind.has_antag_datum(/datum/antagonist/rev,TRUE)
 		converter.add_revolutionary(src,FALSE)
 
-	else if(is_servant_of_ratvar(creator))
+	else if(is_servant_of_ratvar(creator, TRUE))
 		add_servant_of_ratvar(current)
+	else if(is_servant_of_ratvar(creator))
+		add_servant_of_ratvar(current, FALSE, FALSE, /datum/antagonist/clockcult/neutered/traitor)
```
Comment (Ghommie) [inline code/datums/mind.dm]:
```suggestion
	else if(is_servant_of_ratvar(creator))
		if(is_servant_of_ratvar(creator, TRUE))
			add_servant_of_ratvar(current)
		else
			add_servant_of_ratvar(current, FALSE, FALSE, /datum/antagonist/clockcult/neutered/traitor)
```

**Review by Ghommie — CHANGES_REQUESTED** [review]
Little logic performance improvement.

File: code/datums/mind.dm (hunk 1 above)
Comment (DeltaFire15) [reply code/datums/mind.dm]: taken care of

File: code/datums/mind.dm (hunk 2 above)
Comment (DeltaFire15) [reply code/datums/mind.dm]: same here

**Comment by TheObserver-sys** [comment]
Sad to see it go, but it'd only be a matter of time before someone pulls the epic NAR'SIE HAS RISEN on the station and ruin many other players rounds pretty hard core.

**Review by Ghommie — APPROVED** [review]

**Pull request merged**